zetta_utils/mazepa_layer_processing/alignment/common.py

```diff
@@ -29,10 +30,18 @@ def translation_adjusted_download(
     idx: VolumetricIndex,
     src: VolumetricLayer,
     field: VolumetricLayer | None,
+    translation_granularity: int = 1,
 ) -> tuple[torch.Tensor, torch.Tensor | None, tuple[int, int]]:
     if field is not None:
         field_data = field[idx]
-        xy_translation = alignment.field.profile_field2d_percentile(field_data)
+        xy_translation_raw = alignment.field.profile_field2d_percentile(field_data)
+        xy_translation = cast(
+            tuple[int, int],
+            tuple(
+                translation_granularity * (e // translation_granularity)
```

Comment:
Probably doesn't make a big difference, but more accurate would be
```suggestion
                translation_granularity * round(e / translation_granularity)
```